src/flow_field.py

```diff
@@ -12,20 +12,44 @@
 
 # See https://floris.readthedocs.io for documentation
 
+from __future__ import annotations
 
+import attr
 import numpy as np
+import numpy.typing as npt
+
+from src.utilities import FromDictMixin, int_attrib, float_attrib, attrs_array_converter
 
 from .grid import Grid
 
 
-class FlowField:
-    def __init__(self, input_dictionary):
-        self.wind_shear = input_dictionary["wind_shear"]
-        self.wind_veer = input_dictionary["wind_veer"]
-        self.wind_speeds = np.array(input_dictionary["wind_speeds"])
-        self.wind_directions = np.array(input_dictionary["wind_directions"])
-        self.reference_wind_height = input_dictionary["reference_wind_height"]
-        self.air_density = input_dictionary["air_density"]
+NDArrayFloat = npt.NDArray[np.float64]
+
+
+@attr.s(auto_attribs=True)
+class FlowField(FromDictMixin):
+    wind_shear: float = float_attrib()
+    wind_veer: float = float_attrib()
+    wind_speeds: NDArrayFloat = attr.ib(converter=attrs_array_converter)
+    wind_directions: NDArrayFloat = attr.ib(converter=attrs_array_converter)
+    reference_wind_height: int = int_attrib()
+    reference_turbine_diameter: float = float_attrib()
```

Comment:
This attribute should be removed from the FlowField. It's included on the Farm and Grid, and it will be further consolidated in the future.